src/systems/skill/chain_combo_skills.rs

```diff
@@ -0,0 +1,916 @@
+use crate::{
+	components::{ComboTrees, ComboTreesRunning, SlotKey, Slots, Track, WaitNext},
+	skill::{Active, Skill, SkillComboTree},
+};
+use bevy::{
+	ecs::{
+		query::{Added, Without},
+		system::{Commands, EntityCommands, Query},
+		world::Mut,
+	},
+	prelude::Entity,
+};
+
+type SkillComboComponents<'a> = (
+	Entity,
+	&'a ComboTrees,
+	Option<&'a ComboTreesRunning>,
+	&'a Track<Skill<Active>>,
+	&'a mut Slots,
+);
+
+type JustStarted = (Added<Track<Skill<Active>>>, Without<WaitNext>);
+
+pub fn chain_combo_skills(
+	mut commands: Commands,
+	mut idle: Query<(Entity, &mut Slots), Added<WaitNext>>,
+	mut newly_active: Query<SkillComboComponents, JustStarted>,
+) {
+	for (agent, mut slots) in &mut idle {
+		let agent = &mut commands.entity(agent);
+		let slots = &mut slots;
+
+		clear_combos(agent, slots);
+	}
+
+	for (agent, combo_trees, combo_trees_running, track, mut slots) in &mut newly_active {
+		let agent = &mut commands.entity(agent);
+		let slots = &mut slots;
+		let trees = get_trees(combo_trees, combo_trees_running, track);
+
+		clear_combos(agent, slots);
+
+		for tree in &trees {
+			update_combos(agent, slots, tree);
+		}
+	}
+}
+
+fn get_trees(
+	combo_trees: &ComboTrees,
+	combo_trees_running: Option<&ComboTreesRunning>,
+	track: &Track<Skill<Active>>,
+) -> Vec<SkillComboTree> {
+	let tree_running = combo_trees_running.and_then(|t| t.0.get(&track.value.data.slot_key));
+	let trees = combo_trees
+		.0
+		.iter()
+		.filter_map(|t| t.get(&track.value.data.slot_key))
+		.collect();
+	let trees = match tree_running {
+		Some(tree) if !tree.next.is_empty() => vec![tree],
+		_ => trees,
+	};
+
+	trees
+		.into_iter()
+		.cloned()
+		.filter(|t| t.skill.name == track.value.name)
+		.collect()
+}
+
+fn clear_combos(agent: &mut EntityCommands, slots: &mut Mut<Slots>) {
+	for slot in &mut slots.0.values_mut() {
+		slot.combo_skill = None;
+	}
+	agent.remove::<ComboTreesRunning>();
+}
+
+fn update_combos(agent: &mut EntityCommands, slots: &mut Mut<Slots>, tree: &SkillComboTree) {
+	let remaining = tree
+		.next
+		.iter()
+		.filter_map(|(slot_key, tree)| update_slot(slots, slot_key, tree));
+
+	agent.insert(ComboTreesRunning(remaining.collect()));
```

Comment:
This will override any previously inserted component with another combo tree. Maybe first collect all the remaining combo trees -> flatten -> register in one `ComboTreesRunning`, which should be able to hold mutliple combo trees itself.

Reply:
redone how chaining works. Not it is always just one tree present.